src/store/segmenter/segmenterSlice.ts

```diff
@@ -73,13 +73,17 @@ export const segmenterSlice = createSlice({
     ) {
       const { model, inputShape } = action.payload;
 
-      availableSegmenterModels.push(model);
-      state.selectedModelIdx = availableSegmenterModels.length - 1;
-
       state.inputShape = inputShape;
 
       if (model.pretrained) {
         state.modelStatus = ModelStatus.Trained;
+        const selectedModelIdx = availableSegmenterModels.findIndex(
+          (m) => m.constructor.name === model.constructor.name
+        );
+        state.selectedModelIdx = selectedModelIdx >= 0 ? selectedModelIdx : 0;
+      } else {
+        availableSegmenterModels.push(model);
+        state.selectedModelIdx = availableSegmenterModels.length - 1;
```

Comment:
This should technically never be called, since we don't support uploading segmenter models currently. I left it just because we have similar code fort the classifier, where we do support upload.

Reply:
One other thing issue is that the `FitSegmenterDialog` auto-populates with whatever model is selected, regardless of trainability. For example, when COCO-SSD is the selected model, it is populated in the architecture drop down and an info box is displays the message "COCO-SSD requires 3 channels!" even though the "input channels" input is set to 3. This may be misleading as users may reason that theres some connection to the channels and the inability to fit the model.

The reason why non-trainable models are appearing is because of the "value" parameter in the `Autocomplete` component. Removing that stops it from being populated but the information about the required channels still appears. 

Similarly, `modelOptions` should be in a `useMemo` or defined outside the component otherwise a new object is instantiated on each render.

Reply:
I've changed the fit segmenter to mirror the changes I made to fit classifier, where these some of these issues, eg "value" parameter in `AutoComplete`, were originally solved. The loaded pretrained model still appears in the dropdown, but it's now intentional. Fitting is disabled, but now switching from a loaded model like COCO, to FullyConvolutional will ask the user whether they want to dispose first. If they click yes, it will be removed from the dropdown, otherwise it will still be available as an option. Admittedly a bit confusing, but it should stay until we create another UI component that allows you to dispose loaded models.

> For example, when COCO-SSD is the selected model, it is populated in the architecture drop down and an info box is displays the message "COCO-SSD requires 3 channels!" even though the "input channels" input is set to 3. 

That was a feature we intentionally added a while back. I agree, it's confusing and it should probably be either re-worded or changed somehow. The idea is that even for pretrained models that are trainable, the number of channels is often fixed. With trainable models, the message is trying to inform the user that even though the model is trainable, and all options adjustable, the number of channels can't be adjusted, ie that it's set to `3` and can't be re-set, not that it's incorrectly set to `3` when it expects `3`.

> This may be misleading as users could reason that theres some connection between the channels and the inability to fit the model (greyed out fit segmentor button).

I've changed it so that when you hover over the disabled "Fit Segmenter" button, it's a differentiates whether the button is disabled because of "no labeled images" vs "no trainable". It will still likely cause confusion.